src/common/NetworkPrivate.cpp

```diff
@@ -54,360 +149,311 @@ QString NetworkData::getHash()
+                [cb = std::move(this->onSuccess_), result = std::move(result),
+                 hasCaller = this->hasCaller_, caller = this->caller_]() {
+                    if (hasCaller && caller.isNull())
                     {
-                        return NetworkManager::accessManager.sendCustomRequest(
-                            data->request_, "PATCH", data->payload_);
+                        return;
                     }
-            }
-            return nullptr;
-        }();
 
-        if (reply == nullptr)
-        {
-            qCDebug(chatterinoCommon) << "Unhandled request type";
-            return;
-        }
+                    cb(result);
+                });
+}
 
-        if (data->timer_ != nullptr && data->timer_->isActive())
-        {
-            QObject::connect(
-                data->timer_, &QTimer::timeout, worker, [reply, data]() {
-                    qCDebug(chatterinoCommon) << "Aborted!";
-                    reply->abort();
-                    qCDebug(chatterinoHTTP)
-                        << QString("%1 [timed out] %2")
-                               .arg(networkRequestTypes.at(
-                                        int(data->requestType_)),
-                                    data->request_.url().toString());
-
-                    if (data->onError_)
+void NetworkData::emitError(NetworkResult &&result)
+{
+    if (!this->onError_)
+    {
+        return;
+    }
+
+    runCallback(this->executeConcurrently_,
+                [cb = std::move(this->onError_), result = std::move(result),
+                 hasCaller = this->hasCaller_, caller = this->caller_]() {
+                    if (hasCaller && caller.isNull())
                     {
-                        postToThread([data] {
-                            data->onError_(NetworkResult(
-                                NetworkResult::NetworkError::TimeoutError, {},
-                                {}));
-                        });
+                        return;
                     }
 
-                    if (data->finally_)
+                    cb(result);
+                });
+}
+
+void NetworkData::emitFinally()
+{
+    if (!this->finally_)
+    {
+        return;
+    }
+
+    runCallback(this->executeConcurrently_,
+                [cb = std::move(this->finally_), hasCaller = this->hasCaller_,
+                 caller = this->caller_]() {
+                    if (hasCaller && caller.isNull())
                     {
-                        postToThread([data] {
-                            data->finally_();
-                        });
+                        return;
                     }
+
+                    cb();
                 });
-        }
+}
 
-        if (data->onReplyCreated_)
-        {
-            data->onReplyCreated_(reply);
-        }
+QLatin1String NetworkData::typeString() const
+{
+    auto view = magic_enum::enum_name<NetworkRequestType>(this->requestType_);
+    return QLatin1String{view.data(),
```

Comment:
Why is this latin1? Do compilers actually define typenames to be latin1, not ASCII or any of the popular Unicode encodings like UTF-8?

Reply:
> Why is this latin1?

Because it's the only way of printing a string-view over 8bit character data in Qt <6.5 (with 6.5 you can print `std::basic_string_view`). Otherwise, we'd need to create a `QString`.

> Do compilers actually define typenames to be latin1, not ASCII or any of the popular Unicode encodings like UTF-8?

It's a subset of Unicode (see [P1949R7](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p1949r7.html)).

We can't use `QUtf8StringView`, because Qt 5 is supported. But realistically, Latin-1 is totally fine. The names we use for enum values are in the range `A-Z`, `a-z`, `0-9` (and possibly `_`) - these are all in US-ASCII and thus Latin-1.